cmake/multi_image.cmake

```diff
@@ -125,6 +125,22 @@ function(add_child_image_from_source)
   set(${ACI_NAME}_BOARD ${ACI_BOARD})
 
   message("\n=== child image ${ACI_NAME} - ${ACI_DOMAIN}${inherited} begin ===")
+
+  # Look for a folder named 'child_image_overlay' and apply all relevant
+  # overlay files. An overlay file will be applied if it is located in the
+  # root of the child_image_overlay folder and matches ${ACI_NAME}-overlay.conf
+  # or if it is stored in
+  # 'child_image_overlay/parent_board_name/${ACI_NAME}-overlay.conf'
+  set(child_conf ${CMAKE_SOURCE_DIR}/child_image_overlay)
+  foreach (overlay_file
+      ${child_conf}/${ACI_NAME}-overlay.conf
```

Comment:
Based on existing Zephyr usage, we should support:
```
<sample>
  |--- prj.conf
  |--- prj_<desc>.conf
  |--- boards
  |      |--- <board>.conf
  |      |--- <board>_<desc>.conf
  |--- child_image
         |--- <ACI_name>  <-- This is a folder
                |--- prj.conf         <-- If present, use instead of sample/<ACI_name>/prj.conf
                |--- prj_<desc>.conf  <-- If present, used when user specify `-DCONF_FILE=prj_<desc>.conf for parent image instead of sample/<ACI_name>/prj_<desc>.conf
                |--- <ACI_name>_prj.conf         <-- Fragment, use together with sample/<ACI_name>/prj.conf
                |--- <ACI_name>_prj_<desc>.conf  <-- Fragment, use together with sample/<ACI_name>/prj_<desc>.conf
                         |--- boards  
                                |--- <board>.conf          <--- If present, use instead  of sample/<ACI_name>/boards/<board>.conf
                                |--- <board>_<desc>.conf   <--- If present, use instead  of sample/<ACI_name>/boards/<board>_<desc>.conf

```

Reply:
Thanks for the detailed description, I will add this.

Reply:
One more note to this, placing a:
```<sample>
  |--- child_image
         |--- <ACI_name>  
                |--- prj.conf
```
would is an opt-out of `prj.conf` in `samples/<ACI_name>/prj.conf`, and therefore any `samples/<ACI_name>/boards/<board>.conf` should also be ignored.
If user wants a board fragment config, then he should place a corresponding `child_image/<ACI_name>/boards/<board>conf`, like:
```
  |--- child_image
         |--- <ACI_name>  
                |--- prj.conf
                        |--- boards  
                                |--- <board>.conf  
```

Reason for that, is that if we start mixing `<ACI_name>/prj.conf` with `samples/<ACI_name>/boards/<board>.conf` makes it less clear how to opt-out of the board fragment config.
And it is easier to understand that if you have `<ACI_name>/prj.conf`, then all `prj.conf` releated fragments are taken from `child_image/<ACI_NAME>`.

Reply:
as well as any form of `<ACI_name>_prj.conf` or `<ACI__name>_<board>.conf` is an addition to the default `samples/<ACI_name>/` configuration files.

Reply:
I've changed it to work this way now, it should work the way you described.